### PR TITLE
fix: address ambiguities in RLP spec description

### PR DESCRIPTION
**Pull request by prestwich**
Clarifies handling of positive integers in the RLP specification document by updating the definition and description sections

drive-by change: specify that strings of length >= 2**64 bytes may not be encoded

### CODE REVIEW

File: src/content/developers/docs/data-structures-and-encoding/rlp/index.md [hunk 1]
```diff
@@ -20,21 +20,25 @@ The RLP encoding function takes in an item. An item is defined as follows：
...
 
 RLP encoding is defined as follows:
 
+- For a positive integer, it is converted to the the shortest byte array whose big-endian interpretation is the integer, and then encoded as a string according to the rules below.
 - For a single byte whose value is in the `[0x00, 0x7f]` (decimal `[0, 127]`) range, that byte is its own RLP encoding.
 - Otherwise, if a string is 0-55 bytes long, the RLP encoding consists of a single byte with value **0x80** (dec. 128) plus the length of the string followed by the string. The range of the first byte is thus `[0x80, 0xb7]` (dec. `[128, 183]`).
 - If a string is more than 55 bytes long, the RLP encoding consists of a single byte with value **0xb7** (dec. 183) plus the length in bytes of the length of the string in binary form, followed by the length of the string, followed by the string. For example, a 1024 byte long string would be encoded as `\xb9\x04\x00` (dec. `185, 4, 0`) followed by the string. Here, `0xb9` (183 + 2 = 185) as the first byte, followed by the 2 bytes `0x0400` (dec. 1024) that denote the length of the actual string. The range of the first byte is thus `[0xb8, 0xbf]` (dec. `[184, 191]`).
```
Comment (recmo) [inline src/content/developers/docs/data-structures-and-encoding/rlp/index.md:40]: > length of the string in binary form

This is also ambiguous and should specify endianess and leading-zeros treatment. Same with length of the list below.

Comment (prestwich) [reply src/content/developers/docs/data-structures-and-encoding/rlp/index.md:40]: Yeah, there are a number of other fixes that could be made, but I wanted to limit this PR to clarifying the specific weirdness wrt RLP definition of positive integer encoding leaking into higher-level protocols 😮‍💨

**Review by wackerow — APPROVED** [review]
Thanks @prestwich! Sorry for delay here, bringing this in, appreciate the adjustments

**Comment by wackerow** [comment]
@all-contributors please add @prestwich for content

**Pull request merged**